quinn-proto/src/connection.rs

```diff
@@ -1917,12 +1867,22 @@ impl Connection {
     }
 
     fn make_close(&mut self, reason: &state::CloseReason) -> Box<[u8]> {
-        let number = self.get_tx_number();
+        let full_number = self.get_tx_number();
+        let number = PacketNumber::new(full_number, self.largest_acked_packet);
         let mut buf = Vec::new();
-        let header = Header::Short {
-            dst_cid: self.rem_cid,
-            number: PacketNumber::new(number, self.largest_acked_packet),
-            key_phase: self.key_phase,
+        let header = if self.crypto.is_some() {
```

Comment:
Hmm, seems like it would be clearer/more robust to check for `Established` state, instead of `crypto.is_some()`?

Reply:
This is called when the state is already `Closed` (or would be, if it wasn't temporarily removed from the `Option`). Checking `crypto.is_some()` therefore helps ensure we generate a packet that the makes sense to the peer.